src/repositories/user.repository.ts

```diff
@@ -0,0 +1,27 @@
+import {IUserRepository} from "./user.repository.interface";
+import {Repository} from "typeorm/repository/Repository";
+import {User} from "../entity/user.entity";
+import {v4 as uuidv4} from "uuid";
+import {dataSource} from "../dataSource";
+
+export class UserRepository implements IUserRepository {
+    constructor(private repository: Repository<User>) {}
+
+   async getUserByEmail(email: string): Promise<User> {
```

Comment:
don't use word User in names of your repository methods because it is clear that all methods in UserRepository work with User entity